python_modules/dagster-contrib/dagster_contrib/pandas/__init__.py

```diff
@@ -93,40 +84,25 @@ def materialize_runtime_value(self, config_spec, runtime_value):
             check.failed('Unsupported file_type {file_type}'.format(file_type=file_type))
         check.failed('must implement')
 
-    def create_serializable_type_value(self, value, output_dir):
-        check.str_param(output_dir, 'output_dir')
-        csv_path = os.path.join(output_dir, 'csv')
-        value.to_csv(csv_path, index=False)
-        df_meta = DataFrameMeta(format='csv', path='csv')
-        import dagster.core.types.base
-
-        return dagster.core.types.base.SerializedTypeValue(name=self.name, value=df_meta._asdict())
-
-    def deserialize_from_type_value(self, type_value, output_dir):
-        check.str_param(output_dir, 'output_dir')
 
-        df_meta_dict = type_value.value
-        check.inst(df_meta_dict, dict)
-        df_meta = DataFrameMeta(**df_meta_dict)
-
-        if df_meta.format == 'csv':
-            csv_path = os.path.join(output_dir, df_meta.path)
-            return pd.read_csv(csv_path)
-        else:
-            raise Exception('unsupported')
+class DataFrame(FileMarshalable, types.PythonObjectType):
+    def __init__(self):
+        super(DataFrame, self).__init__(
+            name='PandasDataFrame',
+            python_type=pd.DataFrame,
+            description='''Two-dimensional size-mutable, potentially heterogeneous
+    tabular data structure with labeled axes (rows and columns). See http://pandas.pydata.org/''',
+            input_schema_cls=PandasDataFrameInputSchema,
+            output_schema_cls=PandasDataFrameOutputConfigSchema,
+        )
 
-    def construct_from_config_value(self, config_value):
-        file_type, file_options = list(config_value.items())[0]
-        if file_type == 'csv':
-            path = file_options['path']
-            del file_options['path']
-            return pd.read_csv(path, **file_options)
-        elif file_type == 'parquet':
-            return pd.read_parquet(file_options['path'])
-        elif file_type == 'table':
-            return pd.read_table(file_options['path'])
-        else:
-            check.failed('Unsupported file_type {file_type}'.format(file_type=file_type))
+    def marshal_value(self, value, to_file):
```

Comment:
I feel like this should probably be the default implementation on `FileMarshalable`